agent/agent.go

```diff
@@ -603,14 +605,48 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 
 func (a *Agent) endTaskResponse(tc *taskContext, status string, message string) *apimodels.TaskEndDetail {
 	var description string
-	var cmdType string
+	var failureType string
+	if a.endTaskResp != nil { // if the user indicated a task response, use this instead
+		tc.logger.Task().Infof("Task status set using /task_status")
+		if !evergreen.IsValidTaskEndStatus(a.endTaskResp.Status) {
+			tc.logger.Task().Errorf("'%s' is not a valid task status", a.endTaskResp.Status)
+			status = evergreen.TaskFailed
+			failureType = evergreen.CommandTypeSystem
+			tc.logger.Task().Debugf("END TASK RESP: %v", a.endTaskResp)
+		} else {
+			status = a.endTaskResp.Status
+			if len(a.endTaskResp.Description) > MessageLimit {
```

Comment:
How do we intend users to use `Description` versus `Message`?

Reply:
Honestly good question. It looks like the only time we ever set message is [here](https://github.com/evergreen-ci/evergreen/blob/67d01e997b9c444325723bb4fc6a7e541eb0a785/agent/agent.go#L443) and I'm not sure if we display it in the UI or not. We could instead just disallow setting this and not provide in the documentation to users.